e2e-test/cbas_azure_e2e_test.py

```diff
@@ -0,0 +1,220 @@
+from workspace_helper import create_workspace
+from app_helper import create_app, poll_for_app_url
+import requests
+import os
+import json
+import random
+import string
+import uuid
+import time
+
+
+# Setup configuration
+# These values should be injected into the environment before setup
+azure_token = os.environ.get("AZURE_TOKEN")
+bee_name = os.environ.get("BEE_NAME")
+billing_project_name = os.environ.get("BILLING_PROJECT_NAME")
+
+rawls_url = f"https://rawls.{bee_name}.bee.envs-terra.bio"
+leo_url = f"https://leonardo.{bee_name}.bee.envs-terra.bio"
+
+
+# Upload data to WDS using APIs
+def upload_wds_data_using_api(wds_url, workspace_id, tsv_file_name, record_name):
+    #open TSV file in read mode
+    tsv_file = open(tsv_file_name, "r")
+    request_file = tsv_file.read()
+    tsv_file.close()
```

Comment:
I think it's generally better practice to use a context manager (`with open(...)`) instead of opening and closing the manually. (kind of a nit, but it eliminates the risk of that `tsv_file.close()` call being accidentally removed or relocated). Something like:

```
with open(tsv_file_name) as tsv_file:
    request_file = tsv_file.read()
```

Reply:
Ah I didn't know about this. I have updated it in both places. Thank you!